### DISCUSSION TITLE
Centered, Justified, Right, Left Aligned image Style not working in Joomla 4 and 5 editor

### DISCUSSION THREAD

**Discussion by cmirnow** [eworkers/K2ForJ4#59] [Show and tell]
This error occurs in K2ForJ4 articles and in Joomla articles. It is probably a bug in the Joomla editor. I see the problem on several Joomla sites, but I have not been able to convince Joomla developers and maintainers that the bug exists.

Is it possible that I am mistaken? If you see a similar error, please let me know about it. As I see it, the problem manifests itself on any Joomla templates: the css class works strangely in the Joomla backend, but doesn't work on the frontend of the website.

https://github.com/joomla/joomla-cms/issues/42813

**Comment by eworkers** [comment]
maybe it is related to your custom template, however i think you can easily overcome this by adding some rules to your template css's file.

something like this:

```
.align-right img {
  margin-left: auto;
  margin-right: inherit;
}
.align-left img {
  margin-left: inherit;
  margin-right: auto;
}
.align-center img {
  margin-left: auto;
  margin-right: auto;
}
```

**Comment by cmirnow** [comment]
Of course. It's not a difficult problem to solve. But I think it's a bug.

![align](https://github.com/eworkers/K2ForJ4/assets/10660173/8c1307c5-8cab-48fe-a5e1-e40d7234536c)


Take a look at this. I don't understand what this has to do with template overrides. The error shows itself in the text editor, in the default template Cassiopeia, on a newly installed Joomla, where there are no extensions from third-party developers.

Why does the text editor writes the wrong classes?

@chmst :
> This should be text-center, not align-center. I there a template override?

**Comment by eworkers** [comment]
I think the issue was with media insertion plugin in joomla, but just tested now on J5.0.3 and I cannot replicate, I have noticed 2 differences
1- n+1 images are no longer wrapped in <figure>
2- alignment are explicitly inserted inline with style property
